### PR TITLE
Catalyst package

### PR DESCRIPTION
**Pull request by sbna**
New package file to install Catalyst as a module.

### CODE REVIEW

**Comment by alalazo** [comment]
Hi @sbna, nice to see you here! The PR is failing on `flake8`. Just for your information, if you want to test your PRs locally before submitting, you can use the commands:
```
$ spack flake8 # Checks for flake8 conformancy
$ spack test # Run unit tests
```

File: var/spack/repos/builtin/packages/catalyst/package.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
+##############################################################################
+# Copyright (c) 2017 Simone Bna, CINECA.
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/catalyst/package.py]: As an aside, I don't know if a PR where you hold the copyright can be accepted here. Summoning @scheibelp or @tgamblin to have more information on this :smile:

Comment (sbna) [reply var/spack/repos/builtin/packages/catalyst/package.py]: Hi @alalazo, thanks for your suggestions. I will contact tgamblin about the copyright issue.

**Comment by olesenm** [comment]
Is catalyst usable if built without any variants (ie, only with `Base`) or is essentials a reasonable thing to have as a default?

**Comment by olesenm** [comment]
Can probably compact the do_stage() a bit:

        # Specify the editions
        editions = ['Base']
        if '+python' in self.spec:
            editions.append('Enable-Python')
        if '+essentials' in self.spec:
            editions.append('Essentials')
        if '+extras' in self.spec:
            editions.append('Extras')
        if '+rendering' in self.spec:
            editions.append('Rendering-Base')

        # Extract the catalyst part
        paraview_dir = os.path.join(self.stage.path, 'ParaView-v' + str(self.version))
        catalyst_script = os.path.join(paraview_dir, 'Catalyst', 'catalyze.py')
        catalyst_source_dir = os.path.abspath(self.root_cmakelists_dir)

        command = ['python', catalyst_script, '-r', paraview_dir]

        for ed in editions:
            command.append('-i')
            command.append(ed)

        command.append('-o')
        command.append(catalyst_source_dir)

for example.

**Comment by jrood-nrel** [comment]
My apologies for my ignorance of Catalyst, but I'm just curious. Is this treating Catalyst as a subset of Paraview? I skimmed the code but am having trouble following. I use Catalyst in my project Nalu, but I am currently keeping my own custom package files that I haven't found a good way to generalize for a pull request yet. For example, I use the Trilinos Ioss adapter as seen here https://github.com/NaluCFD/NaluSpack/blob/master/spack_config/custom_package_files/catalyst-ioss-adapter/package.py which is a dependency of https://github.com/NaluCFD/NaluSpack/blob/master/spack_config/custom_package_files/nalu/package.py . I just have it install Paraview as a dependency, but would I be able to use this to build Catalyst standalone? It takes 8 hours to build Paraview with the Intel compiler in my experience so it would be nice to reduce this time.

**Comment by sbna** [comment]
Hi @jrood-nrel,  
> Is this treating Catalyst as a subset of Paraview?
yes, it uses a script to extract the catalyst files from the paraview source files.

> I just have it install Paraview as a dependency, but would I be able to use this to build Catalyst standalone?
yes, the script compiles the catalyst source files only and generates a catalyst module.

**Comment by sbna** [comment]
Hi @olesenm,

I don't know how much "big" the Base version is. I have not yet tried to use it.

File: var/spack/repos/builtin/packages/catalyst/package.py (hunk 1 above)
Comment (sbna) [reply var/spack/repos/builtin/packages/catalyst/package.py]: Hi @tgamblin, @scheibelp,
is it possible to set the copyright as above in a package.py file of a PR?

Comment (sbna) [reply var/spack/repos/builtin/packages/catalyst/package.py]: @alalazo CI fails with the following message:
The command "pip install --upgrade codecov" failed and exited with 1 during .

Comment (alalazo) [reply var/spack/repos/builtin/packages/catalyst/package.py]: @sbna I restarted the build that errored, and now everything seems fine. 

@tgamblin There's something fishy going on with coverage. Here we have a +0.35% on `core`, and as far as I can see @sbna just committed a package. Pinging @adamjstewart to keep him in the loop.

**Comment by adamjstewart** [comment]
Discussion seems to have died down, so I'll merge this PR now. If anyone encounters any problems with the package, feel free to open up a new PR.

**Pull request merged**